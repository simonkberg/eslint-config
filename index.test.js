/* eslint-env jasmine */

const path = require('path')
const config = require('./index.js')
const CLIEngine = require('eslint').CLIEngine

describe('@simonkberg/eslint-config', () => {
  it('should match the snapshot', () => {
    expect(
      Object.assign({}, config, {
        extends: config.extends.map(preset => path.relative(__dirname, preset)),
      })
    ).toMatchSnapshot()
  })

  it('should load config in eslint', () => {
    const cli = new CLIEngine({
      useEslintrc: false,
      configFile: './index.js',
    })

    const stub = 'const foo = 1\nconst bar = baz => baz\nbar(foo)\n'
    const result = cli.executeOnText(stub)

    expect(result.errorCount).toEqual(0)
    expect(result.warningCount).toEqual(0)
  })
})
