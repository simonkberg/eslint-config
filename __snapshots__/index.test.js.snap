// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@simonkberg/eslint-config should match the snapshot 1`] = `
Object {
  "extends": Array [
    "node_modules/eslint-config-standard/index.js",
    "node_modules/eslint-config-prettier/index.js",
  ],
}
`;
